browser/ai_chat/ai_chat_browsertests.cc

```diff
@@ -154,4 +160,52 @@ IN_PROC_BROWSER_TEST_F(AIChatBrowserTest, YoutubeNavigations) {
   EXPECT_EQ("Navigated content", navigated_content);
 }
 
+// Test for https://github.com/brave/brave-browser/issues/47294
+IN_PROC_BROWSER_TEST_F(AIChatBrowserTest,
+                       ClosingMultiAssociatedChatDoesNotCrash) {
+  auto get_associated_content = [](content::RenderFrameHost* rfh) {
+    return AIChatTabHelper::FromWebContents(
+        content::WebContents::FromRenderFrameHost(rfh));
+  };
+  auto* content1 =
+      get_associated_content(ui_test_utils::NavigateToURLWithDisposition(
+          browser(), GURL("https://example.com/one"),
+          WindowOpenDisposition::NEW_FOREGROUND_TAB,
+          ui_test_utils::BROWSER_TEST_WAIT_FOR_LOAD_STOP));
+  auto* content2 =
+      get_associated_content(ui_test_utils::NavigateToURLWithDisposition(
+          browser(), GURL("https://example.com/two"),
+          WindowOpenDisposition::NEW_FOREGROUND_TAB,
+          ui_test_utils::BROWSER_TEST_WAIT_FOR_LOAD_STOP));
+  auto* content3 =
+      get_associated_content(ui_test_utils::NavigateToURLWithDisposition(
+          browser(), GURL("https://example.com/three"),
+          WindowOpenDisposition::NEW_FOREGROUND_TAB,
+          ui_test_utils::BROWSER_TEST_WAIT_FOR_LOAD_STOP));
+
+  auto* ai_chat_service =
+      AIChatServiceFactory::GetForBrowserContext(browser()->profile());
+  auto* conversation = ai_chat_service->CreateConversation();
+  ai_chat_service->MaybeAssociateContent(content1,
+                                         conversation->get_conversation_uuid());
+  ai_chat_service->MaybeAssociateContent(content2,
+                                         conversation->get_conversation_uuid());
+  ai_chat_service->MaybeAssociateContent(content3,
+                                         conversation->get_conversation_uuid());
+
+  EXPECT_EQ(ai_chat_service->GetInMemoryConversationCountForTesting(), 1u);
+
+  ui_test_utils::NavigateToURLWithDisposition(
+      browser(),
+      GURL(base::StrCat(
+          {"chrome://leo-ai/", conversation->get_conversation_uuid()})),
+      WindowOpenDisposition::NEW_FOREGROUND_TAB,
+      ui_test_utils::BROWSER_TEST_WAIT_FOR_LOAD_STOP);
+
+  base::RunLoop().Run();
```

Comment:
why do we need a runloop Run() without pairing with Quit() here?

Reply:
Whoops, that shouldn't have been committed - I tested with --enable-pixel-output to make sure it was doing what I thought :laughing: